conans/model/editable_cpp_info.py

```diff
@@ -0,0 +1,57 @@
+# coding=utf-8
+import configparser
+import ntpath
+import os
+import posixpath
+import re
+
+import six
+
+
+class EditableCppInfo(object):
+    cpp_info_dirs = ['includedirs', 'libdirs', 'resdirs', 'bindirs']
+
+    def __init__(self, data):
+        self._data = data
+
+    @classmethod
+    def create(cls, filepath_or_content, *args, **kwargs):
+        _parse_func = cls.parse_file if os.path.exists(filepath_or_content) else cls.parse_content
+        data = _parse_func(filepath_or_content, *args, **kwargs)
+        return EditableCppInfo(data)
+
+    @classmethod
+    def parse_file(cls, filepath, base_path, settings=None, options=None):
+        with open(filepath, 'r') as f:
+            return cls.parse_content(content=six.u(f.read()), base_path=base_path,
+                                     settings=settings, options=options)
+
+    @classmethod
+    def parse_content(cls, content, base_path, settings=None, options=None):
+        """ Returns a dictionary containing information about paths for a CppInfo object: includes,
+        libraries, resources, binaries,... """
+        ret = {k: [] for k in cls.cpp_info_dirs}
+
+        def _work_on_value(value, base_path_, settings_, options_):
+            value = re.sub(r'\\\\+', r'\\', value)
+            value = value.replace('\\', '/')
+            isabs = ntpath.isabs(value) or posixpath.isabs(value)
```

Comment:
This way I can have the same tests for windows and linux, otherwise I have to activate/deactivate depending on OS. Battery of tests is here: https://github.com/conan-io/conan/pull/4181/files#diff-b2d334bb20f76d192915fdaa29673cb7R22. Let's keep this comment open, I may change it

Reply:
Yes, this reads confusing. If testing is the reason, the complexity should be moved somewhere in the test suite. Please consider simplifying it.

Reply:
Writing some unittests I have realized that this is needed: 

If there is a file stored in the package repo, it will contain paths for Windows and Linux, so I need to be sure that the parsing is done exactly the same when running in both systems. 
Nevertheless I should add more logic here: discard Windows paths in Linux and viceversa (some build systems may fail if given malformed paths). Thinking about how to check if a path is valid... 🤔

Reply:
I am not sure about the underlying reason for this. Forward slash paths also work fine in Windows, is it not enough to use that for the layout files? Internally, if necessary, it could be used ``os.path.normpath`` to normalize to whatever every system wants, but needing to check the platform to run different things with the paths seems to much, it was not needed so far, and conan is using paths in multiple places without this problem. I might be still missing something.

Reply:
The user is writing this file, we don't have control over it, he/she can write paths like this:

```
/linux/abs/path
../relative/path
..\also\relative\path
C:\windows\abs\path
C:\\sometimes\\seen\\like\\this
```

I emplace you two to the unittests section (`WorkOnItemsTest`) and let me know which tests you would remove.